app/lib/account/backend.dart

```diff
@@ -180,6 +180,9 @@ class AccountBackend {
       return null;
     }
     final user = await _lookupOrCreateUserByOauthUserId(auth);
+    if (user.isDeleted) {
+      return null;
```

Comment:
This should not be possible, let's write a warning to the logs.
Said log message also serves as documentation of what this is checking for.